addons/osfstorage/views.py

```diff
@@ -282,6 +282,8 @@ def osfstorage_get_children(file_node, **kwargs):
 def osfstorage_create_child(file_node, payload, **kwargs):
     parent = file_node  # Just for clarity
     name = payload.get('name')
+    # If you are uploading a new version of an existing file with a new name.
+    new_name = payload.get('new_name')
```

Comment:
hmm, actually if i'm dragging in a new file (with a new name) and I want to indicate that it needs to be a new version of another file instead of a new file altogether, it's name is actually going to be it's new name.  So the extra variable I pass in should be `former_name` or something? So we look up the file by former_name, if it exists, make the new file a new version of it.

Reply:
From the API, the request is going to be a PATCH to the old file if it's replacing, and a PATCH to the folder if it's intended to be new. From the UI, Fangorn isn't going to handle this particular case, so I think it would be okay just to handle this in a distinct way rather than incorporating this into existing methods. Does that make sense?

Reply:
ok, even better! but I still need to add handling for this somewhere, right?

Reply:
As discussed via zoom, there's nothing that the back end needs to handle specifically related to this. It'll be all either API requests or a modification to Waterbutler.

Reply:
cool, I'm removing these recent changes -